woocommerce/woocommerce
# fix https://github.com/woocommerce/woocommerce/issues/51548

### Submission Review Guidelines:

### Changes proposed in this Pull Request:

Closes #51548 

### How to test the changes in this Pull Request:

1. Create a few products and [designate some of them as upsells of another one ](https://woocommerce.com/document/related-products-up-sells-and-cross-sells/). Take note of the product id.
2. Add the following code snippet to your site:

```php
add_filter('woocommerce_upsell_display_args', function($args) {
    echo "<h1>Limit is a " . getType($args['posts_per_page']) . " with value " . $args['posts_per_page'] . "</h1>";
    return $args;
});
```

3. Run the following in CLI replacing the product id as appropriate, you should see `<h1>Limit is a integer with value -1</h1>` printed in the terminal:

```
wp eval 'global $product;$product=wc_get_product(<product id>);woocommerce_upsell_display();'
```

### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Fix: woocommerce_upsell_display was receiving a string instead of an integer in $limit

</details>
